python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,232 @@
+
+            return _GinBuilderInfo(sig_def=sig_def)
+
+        return _GinBuilder(import_graph_fn).build(feed_names, fetch_names)
+
+    @classmethod
+    def _from_saved_model_impl(cls, saved_model_dir, tag_set,
+                               signature_def_key=None,
+                               feed_names=None,
+                               fetch_names=None):
+        """
+        Construct a TFInputGraphBuilder from a SavedModel
+        """
+        assert (feed_names is None) == (fetch_names is None), \
+            'feed_names and fetch_names, if provided must appear together'
+        assert (feed_names is None) != (signature_def_key is None), \
+            'must either provide feed_names or singnature_def_key'
+
+        def import_graph_fn(sess):
+            tag_sets = tag_set.split(',')
+            meta_graph_def = tf.saved_model.loader.load(sess, tag_sets, saved_model_dir)
+
+            sig_def = None
+            if signature_def_key is not None:
+                sig_def = tf.contrib.saved_model.get_signature_def_by_key(
+                    meta_graph_def, signature_def_key)
+
+            return _GinBuilderInfo(sig_def=sig_def)
+
+        return _GinBuilder(import_graph_fn).build(feed_names, fetch_names)
+
+
+class _GinBuilderInfo(object):
+    def __init__(self, sig_def=None):
+        self.sig_def = sig_def
+        self.feed_names = None
+        self.feed_mapping = None
+        self.fetch_names = None
+        self.fetch_mapping = None
+
+    def extract_signatures(self):
+        assert self.sig_def is not None, \
+            "ask to find sigdef mapping, but not found any"
+
+        self.feed_mapping = {}
+        self.feed_names = []
+        for sigdef_key, tnsr_info in self.sig_def.inputs.items():
+            tnsr_name = tnsr_info.name
+            self.feed_mapping[sigdef_key] = tnsr_name
+            self.feed_names.append(tnsr_name)
+
+        self.fetch_mapping = {}
+        self.fetch_names = []
+        for sigdef_key, tnsr_info in self.sig_def.outputs.items():
+            tnsr_name = tnsr_info.name
+            self.fetch_mapping[sigdef_key] = tnsr_name
+            self.fetch_names.append(tnsr_name)
+
+class _GinBuilder(object):
+    def __init__(self, import_graph_fn, sess=None, graph=None):
+        self.import_graph_fn = import_graph_fn
+        assert (sess is None) == (graph is None)
+        if sess is not None:
+            self.graph = graph
+            self.sess = sess
+            self._should_clean = False
+        else:
+            self.graph = tf.Graph()
+            self.sess = tf.Session(graph=self.graph)
+            self._should_clean = True
+
+    def _build_impl(self, feed_names, fetch_names):
+        # pylint: disable=protected-access,attribute-defined-outside-init
+        gin = TFInputGraph._new_obj_internal()
+        assert (feed_names is None) == (fetch_names is None)
+        must_have_sig_def = fetch_names is None
```

Comment:
don't need this variable since it's only used once. just use `if fetch_names is None` where we use this variable below.

Reply:
I think it is easier to reason with when assigned to a variable. For the time being `fetch_name is None` is the only requirement, but it might change in the future. And if it does change, one would only have to modify it here.

Reply:
if it changes in the future we can use a variable.

Reply:
Removed the boolean variable